app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -0,0 +1,108 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class Parse {
+    public HashMap<String, Contents> constRepo = new HashMap<>();
+    public HashMap<String, ArrayList> structRepo = new HashMap<>();
+
+    public Parse(String code) throws Exception {
+        IASTTranslationUnit translationUnit = getIASTTranslationUnit(code.toCharArray());
+        process(translationUnit);
+    }
+
+    private void process(IASTNode node) throws Exception {
+        IASTNode[] children = node.getChildren();
+        for (IASTNode child : children) {
+            if (child instanceof CPPASTSimpleDeclaration) {
+                CPPASTSimpleDeclaration declaration = (CPPASTSimpleDeclaration) child;
+
+                if (declaration.getDeclSpecifier() instanceof CPPASTCompositeTypeSpecifier) {
+                    CPPASTCompositeTypeSpecifier compSpec = (CPPASTCompositeTypeSpecifier) declaration.getDeclSpecifier();
+                    CPPASTName name = (CPPASTName) compSpec.getName();
+                    saveStuct(compSpec.getDeclarations(false), name.getRawSignature());
+                } else {
+                    saveConst(child);
+                }
+
+            }
+        }
+    }
+
+    private ArrayList getComponents(IASTNode node) throws Exception {
+        ArrayList components = new ArrayList();
+        CPPASTSimpleDeclaration declaration = (CPPASTSimpleDeclaration) node;
+        CPPASTNamedTypeSpecifier specifier = (CPPASTNamedTypeSpecifier) declaration.getDeclSpecifier();
+        CPPASTDeclarator declarator = (CPPASTDeclarator) declaration.getDeclarators()[0];
+        CPPASTEqualsInitializer init = (CPPASTEqualsInitializer) declarator.getInitializer();
+
+        if (declarator != null || specifier != null || declaration != null && init != null) {
+            CPPASTName name = (CPPASTName) declarator.getName(); //get name
+            IToken typeQualifier = (IToken) specifier.getSyntax(); //get typeQualifer type (iToken)
+            CPPASTName type = (CPPASTName) specifier.getName(); //get type
+            components.add(name); components.add(typeQualifier); components.add(type); components.add(init);
+        }
+
+        return components;
+    }
+
+    private void saveConst(IASTNode node) throws Exception {
+        ArrayList components = getComponents(node);
+        if (!components.isEmpty()) {
+            CPPASTName name = (CPPASTName) components.get(0);
+            IToken typeQualifier = (IToken) components.get(1);
+            CPPASTName type = (CPPASTName) components.get(2);
+            CPPASTEqualsInitializer init = (CPPASTEqualsInitializer) components.get(3);
+            CPPASTLiteralExpression value = (CPPASTLiteralExpression) init.getInitializerClause();
+            Contents contents = new Contents(name, value, typeQualifier, type);
+            constRepo.put(name.getRawSignature(), contents);
+        }
+    }
+
+    private void saveStrucMem(IASTNode node, ArrayList struct) throws Exception {
```

Comment:
Consider putting this directly into saveStruct? This isn't called anywhere else, and I don't think this give you any significant abstraction / readability benefits?